src/rpc/mining.cpp

```diff
@@ -218,7 +218,7 @@ UniValue generate(const UniValue& params, bool fHelp)
         CBlock *pblock = &pblocktemplate->block;
         {
             LOCK(cs_main);
-            IncrementExtraNonce(pblock, chainActive.Tip(), nExtraNonce);
+            IncrementExtraNonce(pblocktemplate.get(), chainActive.Tip(), nExtraNonce, Params().GetConsensus());
```

Comment:
Why does this need to change?

Reply:
Because if this function is called, the coinbase transaction changes, so `hashAuthDataRoot` changes, and `hashBlockCommitments` needs to be recalculated. We don't store all of the block commitments in the block header, so we use the block template to cache them.